pkg/services/oracle/jsonpath/jsonpath.go

```diff
@@ -0,0 +1,476 @@
+			if err != nil {
+				return nil, false
+			}
+
+			return p.descentIndex(objs, int(index))
+		default:
+			return nil, false
+		}
+	case pathString:
+		subTyp, _, ok := p.nextToken()
+		if !ok {
+			return nil, false
+		}
+
+		switch subTyp {
+		case pathComma:
+			return p.processUnion(objs, pathString, value)
+		case pathRightBracket:
+			s := strings.Trim(value, "'")
+			err := json.Unmarshal([]byte(`"`+s+`"`), &s)
+			if err != nil {
+				return nil, false
+			}
+			return p.descentIdent(objs, s)
+		default:
+			return nil, false
+		}
+	default:
+		return nil, false
+	}
+}
+
+// processUnion processes union of multiple indices.
+// firstTyp is assumed to be either pathNumber or pathString.
+func (p *pathParser) processUnion(objs []interface{}, firstTyp pathTokenType, firstVal string) ([]interface{}, bool) {
+	items := []string{firstVal}
+	for {
+		typ, val, ok := p.nextToken()
+		if !ok || typ != firstTyp {
+			return nil, false
+		}
+
+		items = append(items, val)
+		typ, val, ok = p.nextToken()
+		if !ok {
+			return nil, false
+		} else if typ == pathRightBracket {
+			break
+		} else if typ != pathComma {
+			return nil, false
+		}
+	}
+
+	switch firstTyp {
+	case pathNumber:
+		values := make([]int, len(items))
+		for i := range items {
+			index, err := strconv.ParseInt(items[i], 10, 32)
+			if err != nil {
+				return nil, false
+			}
+			values[i] = int(index)
+		}
+		return p.descentIndex(objs, values...)
+	case pathString:
+		for i := range items {
+			s := strings.Trim(items[i], "'")
+			err := json.Unmarshal([]byte(`"`+s+`"`), &items[i])
+			if err != nil {
+				return nil, false
+			}
+		}
+		return p.descentIdent(objs, items...)
+	default:
+		panic("token in union must be either number or string")
+	}
+}
+
+// processSlice processes slice with the specified start index.
+func (p *pathParser) processSlice(objs []interface{}, start int) ([]interface{}, bool) {
+	typ, val, ok := p.nextToken()
+	if !ok {
+		return nil, false
+	}
+
+	switch typ {
+	case pathNumber:
+		typ, _, ok := p.nextToken()
+		if !ok || typ != pathRightBracket {
+			return nil, false
+		}
+
+		index, err := strconv.ParseInt(val, 10, 32)
+		if err != nil {
+			return nil, false
+		}
+
+		return p.descentRange(objs, start, int(index))
+	case pathRightBracket:
+		return p.descentRange(objs, start, 0)
+	default:
+		return nil, false
+	}
+}
+
+// descentRange is similar to descent but skips maps and returns sub-slices for arrays.
+func (p *pathParser) descentRange(objs []interface{}, start, end int) ([]interface{}, bool) {
+	if p.depth <= 0 {
+		return nil, false
+	}
+	p.depth--
+
+	var values []interface{}
+	for i := range objs {
+		arr, ok := objs[i].([]interface{})
+		if !ok {
+			continue
+		}
+
+		subStart := start
+		if subStart < 0 {
+			subStart += len(arr)
+		}
+
+		subEnd := end
+		if subEnd <= 0 {
+			subEnd += len(arr)
+		}
+
+		if subEnd > len(arr) {
+			subEnd = len(arr)
+		}
+
+		count := subEnd - subStart
+		if count < 0 {
+			return nil, false
```

Comment:
And I think C# returns `true` in this case, https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.take?view=net-5.0

> If count is less than or equal to zero, source is not enumerated and an empty IEnumerable<T> is returned.

Reply:
Yeah, tested this again, fiexd.